### ISSUE TITLE
Synctex pointing to AppData directory

### ISSUE THREAD

**Issue by Tobii42** [James-Yu/Atom-LaTeX#27]
Hi,

when using atom-latex with synctex (which is in my PATH) on windows, I have the following behavior:

Clicking with Ctrl + Left Mouse in the PDF, synctex sometimes does nothing at all. But sometimes, it opens a file with the correct name but not in my project directory, but in the `AppData\Local\atom\app-1.15.0\` directory. The file there is empty.

Synctex the other way around (Right click in tex file and `Synctec from here`) does sometimes work and sometimes does nothing.

Thank you for your time!

Greetings,
Tobias

**Comment by James-Yu** [comment]
That is very strange. Atom-LaTeX will parse the output of SyncTeX and use the output file location to open tex files. Sources are
https://github.com/James-Yu/Atom-LaTeX/blob/master/lib/locator.coffee#L56
https://github.com/James-Yu/Atom-LaTeX/blob/master/lib/locator.coffee#L62

Can you please open your .synctex.gz file and check if the tex file path recorded in that file is correct? I suspect this is an upstream bug of SyncTeX.

**Comment by Tobii42** [comment]
For context: I have splitted the LaTeX document into separate files and use the input-command to put all together: `\input{content/1_introduction.tex}`, ...

The beginning of the synctex file looks like the following:

```
SyncTeX Version:1
Input:1:vldb.tex
Input:2:vldb.cls
Input:3:C:\Program Files\MiKTeX 2.9\tex\latex\geometry\geometry.sty
[...]
Input:38:C:\Program Files\MiKTeX 2.9\tex\latex\hyperref\hpdftex.def
Input:39:C:\Program Files\MiKTeX 2.9\tex\latex\oberdiek\rerunfilecheck.sty
Input:40:vldb.aux
[...]
Input:44:vldb.out
Input:45:vldb.out
[...]
Input:49:content/0_abstract.tex
Input:50:content/1_introduction.tex
Input:51:content/2_processing.tex
Output:pdf
```
Interesting is that the files where nothing happens are not in that input list. The files where the wrong file is opened (e.g., `AppData\Local\atom\app-1.15.0\content\2_processing.tex`) are in that list (Input 49 to 51). The missing inputs seem to be my or synctex's fault, as it happens when building with TexMaker, too.

When using for example TexMaker to open the PDF I built with atom-latex, synctex works correctly for abstract, introduction, etc., but not for the files that are not in that list (error message: not enough rights to open the file).

**Comment by Tobii42** [comment]
Maybe it is because the atom.exe is in "C:\Users\User\AppData\Local\atom\app-1.15.0\atom.exe" and therefore the relative path "content/1_introduction.tex" is mapped to "C:\Users\User\AppData\Local\atom\app-1.15.0\content\1_introduction.tex"?

**Comment by James-Yu** [comment]
I suspect that this is a problem related to SyncTeX. I opened one of my project and the file paths after `Input:xx:` are always full path:

```
SyncTeX Version:1
Input:1:/home/james/some/path/to/project/./main.tex
Input:2:/usr/share/texlive/texmf-dist/tex/latex/IEEEtran/IEEEtran.cls
Input:3:/usr/share/texlive/texmf-dist/tex/latex/psnfss/ot1ptm.fd
Input:4:/usr/share/texlive/texmf-dist/tex/latex/algorithms/algorithm.sty
[...]
```

What's the version of your SyncTeX? Mine is (linux)

```
This is SyncTeX command line utility, version 1.3
usage: synctex <subcommand> [options] [args]
Synchronize TeXnology command-line client, version 1.18
```

**Comment by Tobii42** [comment]
I've looked into some of my LaTeX-projects and always had these relative paths in the synctex-file. My LaTeX-Version is `This is SyncTeX command line utility, version 1.3` (running on Windows 10). Well, maybe it's just some strange configuration on my machine ...

**Comment by James-Yu** [comment]
Can you please provide some information on your LaTeX compiler? I had been working on Windows 10 with `pdflatex` and had not meet similar problems.

**Comment by Tobii42** [comment]
I am using MikTex (`This is pdfTeX, Version 3.14159265-2.6-1.40.16 (MiKTeX 2.9 64-bit)`). Before atom-latex I was using TexMaker. Synctex was not an issue.

When started using atom-latex, building the synctex-file worked totally fine. But when using a Ctrl + Click in the viewer, the following error occured:

```
Command failed: synctex edit -o "1:199.20000000000002:299.99999999999994:C:\Users\Tobias\Dropbox\Studium\Promotion\GIT\Paper\PhD Workshop\paper\vldb.pdf"
Der Befehl "synctex" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
```
(Last part is German for "The command "synctex" cannot be found".)

So, I searched for a solution and ended up with this: http://tex.stackexchange.com/questions/338078/how-to-get-synctex-for-windows-to-allow-atom-pdf-view-to-synch#comment877274_338117

Hence, I placed the two files in the given directory. It seems (from the answer that the comment belongs to), that other viewers sometimes provide a synctex binary what would explain why TexMaker did not need synctex in PATH.

**Comment by James-Yu** [comment]
Yes, standalone LaTeX software tend to ship with synctex binaries. However, it may cause security issues if they are shipped with third-party extensions. What make things strange is that when I was on Windows 10 with TeX Live full distro I never had the problem. Maybe the binary is tampered?

**Comment by Tobii42** [comment]
Thank you for your support! I will try to find a solution or just use pdf-view as a pdf viewer. With this one, synctex works correctly.

One last error message that maybe is interesting for you as it's based on your last commit:

```
Failed SyncTeX. No file path is given.
[object Object]
```

That is for the part of the document where no file is opened and the error probably is within my synctex version / installation.

Keep up the great work with this package!

**Comment by James-Yu** [comment]
hmmm interesting... Will investigate the issue.

**Comment by James-Yu** [comment]
btw, pdf-view ships with synctex binaries? That would worth checking. Maybe I was wrong and the problem was caused by my code.

**Comment by Tobii42** [comment]
> btw, pdf-view ships with synctex binaries?

No, synctex in pdf-view only works if synctex is in PATH.

**Comment by ashthespy** [comment]
@Tobii42  The Stackexchange comment you link to in fact is me :-) 
I am on similar setup as you
```
Windows 10
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (MiKTeX 2.9.6210 64-bit)

This is SyncTeX command line utility, version 1.3
SyncTeX ERROR: Missing options
usage: synctex <subcommand> [options] [args]
Synchronize TeXnology command-line client, version 1.18
````
I seem however to be unable to reproduce your error, and have complete paths in my `.syntex` file. Are you sure it's not something particular to this particular document that you are compiling?

**Comment by Tobii42** [comment]
@ashthespy The world seems to be small :-)

The setup indeed seems to be similar. I tried to compile standard projects (templates from conferences, minimal LaTeX document with just 5 lines). In all cases, the paths are relative and hence, atom-latex opens up a non-existing file in the AppData folder.

Nevertheless, I won't say that it's not me. Maybe I just have to freshly setup Atom, MikTex, etc. On the other hand, other viewers seem to be able to work with relative paths.

Synctex of minimal document:
```
SyncTeX Version:1
Input:1:test.tex
Input:2:C:\Program Files\MiKTeX 2.9\tex\latex\base\article.cls
Input:3:C:\Program Files\MiKTeX 2.9\tex\latex\base\size10.clo
Input:4:C:\Program Files\MiKTeX 2.9\tex\latex\base\inputenc.sty
Input:5:C:\Program Files\MiKTeX 2.9\tex\latex\base\utf8.def
Input:6:C:\Program Files\MiKTeX 2.9\tex\latex\base\t1enc.dfu
Input:7:C:\Program Files\MiKTeX 2.9\tex\latex\base\ot1enc.dfu
Input:8:C:\Program Files\MiKTeX 2.9\tex\latex\base\omsenc.dfu
Input:9:test.aux
Output:pdf
```

**Comment by ashthespy** [comment]
Interesting! What compiler toolchain are you using? I just tried `pdflatex` , `xelatex` and `arara` - all gave me absolute paths - I did not try `latexmk` however.

**Comment by Tobii42** [comment]
Hi,

thanks a lot for your help! I just took a look at the synctex file of a colleague and he had absolute paths as well. Googling revealed that it seems to have been a discussion if relative or absolute paths are better (https://sourceforge.net/p/itexmac/bugs/136/ , https://sourceforge.net/p/miktex/bugs/2088/ , http://tug.org/pipermail/pdftex/2010-January/008303.html ).

Anyway, I just reinstalled MikTex and now I'm getting absolute paths in my synctex file. Subsequently, atom-latex just works fine as it should. Thank you so much, just learned a lot :)

Greetings
